packages/sdk/src/oracle/services/statemachines/disputePrice.ts

```diff
@@ -36,23 +36,35 @@ export function Handlers(store: Store): GenericHandlers<Params, Memory> {
       const { chainId, confirmations, checkTxIntervalSec } = params;
       const { hash } = memory;
       assert(hash, "requires hash");
-      if (await update.isConfirmed(chainId, hash, confirmations)) {
+      const receipt = await update.isConfirmed(chainId, hash, confirmations);
+      if (receipt) {
+        memory.receipt = receipt as TransactionReceipt;
         return "update";
       }
       // wait x seconds before running this state again
       return context.sleep(checkTxIntervalSec * 1000);
     },
     async update(params: Params, memory: Memory) {
       const { chainId, currency, account, requester, identifier, timestamp, ancillaryData } = params;
-      const { hash } = memory;
+      const { hash, receipt } = memory;
+      const oracle = store.read().oracleService(chainId);
       await update.balance(chainId, currency, account);
       await update.request(params);
-      store.write((w) =>
-        w
-          .chains(chainId)
+      store.write((w) => {
+        w.chains(chainId)
           .optimisticOracle()
-          .request({ chainId, requester, identifier, timestamp, ancillaryData, disputeTx: hash })
-      );
+          .request({ chainId, requester, identifier, timestamp, ancillaryData, disputeTx: hash });
+
+        if (receipt) {
+          const events = receipt.logs.map(oracle.makeEventFromLog);
+          events.forEach((event) => {
+            w.chains(chainId)
+              .optimisticOracle()
+              .event((event as unknown) as OptimisticOracleEvent);
+          });
+        }
+      });
+      await update.sortedRequests(chainId);
```

Comment:
the problem was that when you triggered a transaction for propose or dispute from the detail page, we did not push the events from the tx receipt into the same event list that the index page was using to generate its full list of requests. So thats added here for  both types of transactions.